stellargraph/core/graph.py

```diff
@@ -237,6 +237,69 @@ def out_nodes(self, node: Any) -> Iterable[Any]:
         """
         raise NotImplementedError
 
+    def nodes_of_type(self, node_type=None):
+        """
+        Get the nodes of the graph with the specified node types.
+
+        Args:
+            node_type: a type of nodes that exist in the graph
+
+        Returns:
+            A list of node IDs with type node_type
+        """
+        raise NotImplementedError
+
+    def node_type(self, node):
+        """
+        Get the type of the node
+
+        Args:
+            node: Node ID
+
+        Returns:
+            Node type
+        """
+        raise NotImplementedError
+
+    @property
+    def node_types(self):
+        """
+        Get a list of all node types in the graph.
+
+        Returns:
+            set of types
+        """
+        raise NotImplementedError
+
+    def node_feature_sizes(self, node_types=None):
+        """
+        Get the feature sizes for the specified node types.
+
+        Args:
+            node_types: (list) A list of node types. If None all current node types
+                will be used.
+
+        Returns:
+            A dictionary of node type and integer feature size.
+        """
+        raise NotImplementedError
+
+    def node_features(self, nodes, node_type=None):
+        """
+        Get the numeric feature vectors for the specified node or nodes.
+        If the node type is not specified the node types will be found
+        for all nodes. It is therefore important to supply the ``node_type``
+        for this method to be fast.
+
+        Args:
+            n: (list or hashable) Node ID or list of node IDs
```

Comment:
```suggestion
            nodes: (list or hashable) Node ID or list of node IDs
```

Reply:
Hmm actually I think the type is supposed to come before the colon right?
```suggestion
            n (list or hashable): Node ID or list of node IDs
```
and similarly for below